typescript/schedules.md

```diff
@@ -0,0 +1,1246 @@
+  | 0
+  | 1
+  | 2
+  | 3
+  | 4
+  | 5
+  | 6
+  | 7
+  | 8
+  | 9
+  | 10
+  | 11
+  | 12
+  | 13
+  | 14
+  | 15
+  | 16
+  | 17
+  | 18
+  | 19
+  | 20
+  | 21
+  | 22
+  | 23
+  | 24
+  | 25
+  | 26
+  | 27
+  | 28
+  | 29
+  | 30
+  | 31
+  | 32
+  | 33
+  | 34
+  | 35
+  | 36
+  | 37
+  | 38
+  | 39
+  | 40
+  | 41
+  | 42
+  | 43
+  | 44
+  | 45
+  | 46
+  | 47
+  | 48
+  | 49
+  | 50
+  | 51
+  | 52
+  | 53
+  | 54
+  | 55
+  | 56
+  | 57
+  | 58
+  | 59;
+
+export type Second = Range<ZeroTo59> | ZeroTo59;
+export type Minute = Range<ZeroTo59> | ZeroTo59;
+
+export type ZeroTo23 =
+  | 0
+  | 1
+  | 2
+  | 3
+  | 4
+  | 5
+  | 6
+  | 7
+  | 8
+  | 9
+  | 10
+  | 11
+  | 12
+  | 13
+  | 14
+  | 15
+  | 16
+  | 17
+  | 18
+  | 19
+  | 20
+  | 21
+  | 22
+  | 23;
+
+export type Hour = Range<ZeroTo23> | ZeroTo23;
+
+export type ZeroTo30 =
+  | 0
+  | 1
+  | 2
+  | 3
+  | 4
+  | 5
+  | 6
+  | 7
+  | 8
+  | 9
+  | 10
+  | 11
+  | 12
+  | 13
+  | 14
+  | 15
+  | 16
+  | 17
+  | 18
+  | 19
+  | 20
+  | 21
+  | 22
+  | 23
+  | 24
+  | 25
+  | 26
+  | 27
+  | 28
+  | 29
+  | 30;
+
+export type DayOfMonth = Range<ZeroTo30> | ZeroTo30;
+
+export type ZeroTo11 = 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11;
+
+export type MonthName =
+  | 'JAN'
+  | 'JANUARY'
+  | 'FEB'
+  | 'FEBRUARY'
+  | 'MAR'
+  | 'MARCH'
+  | 'APR'
+  | 'APRIL'
+  | 'MAY'
+  | 'JUN'
+  | 'JUNE'
+  | 'JUL'
+  | 'JULY'
+  | 'AUG'
+  | 'AUGUST'
+  | 'SEP'
+  | 'SEPTEMBER'
+  | 'OCT'
+  | 'OCTOBER'
+  | 'NOV'
+  | 'NOVEMBER'
+  | 'DEC'
+  | 'DECEMBER';
+
+export type MonthValue = ZeroTo11 | MonthName;
+
+export type Month = Range<MonthValue> | MonthValue;
+
+export type TwoThousandTo2100 =
+  | 2000
+  | 2001
+  | 2002
+  | 2003
+  | 2004
+  | 2005
+  | 2006
+  | 2007
+  | 2008
+  | 2009
+  | 2010
+  | 2011
+  | 2012
+  | 2013
+  | 2014
+  | 2015
+  | 2016
+  | 2017
+  | 2018
+  | 2019
+  | 2020
+  | 2021
+  | 2022
+  | 2023
+  | 2024
+  | 2025
+  | 2026
+  | 2027
+  | 2028
+  | 2029
+  | 2030
+  | 2031
+  | 2032
+  | 2033
+  | 2034
+  | 2035
+  | 2036
+  | 2037
+  | 2038
+  | 2039
+  | 2040
+  | 2041
+  | 2042
+  | 2043
+  | 2044
+  | 2045
+  | 2046
+  | 2047
+  | 2048
+  | 2049
+  | 2050
+  | 2051
+  | 2052
+  | 2053
+  | 2054
+  | 2055
+  | 2056
+  | 2057
+  | 2058
+  | 2059
+  | 2060
+  | 2061
+  | 2062
+  | 2063
+  | 2064
+  | 2065
+  | 2066
+  | 2067
+  | 2068
+  | 2069
+  | 2070
+  | 2071
+  | 2072
+  | 2073
+  | 2074
+  | 2075
+  | 2076
+  | 2077
+  | 2078
+  | 2079
+  | 2080
+  | 2081
+  | 2082
+  | 2083
+  | 2084
+  | 2085
+  | 2086
+  | 2087
+  | 2088
+  | 2089
+  | 2090
+  | 2091
+  | 2092
+  | 2093
+  | 2094
+  | 2095
+  | 2096
+  | 2097
+  | 2098
+  | 2099
+  | 2100;
+
+/**
+ * Temporal Server currently only supports specifying years in the range `[2000, 2100]`.
+ */
+export type Year = Range<TwoThousandTo2100> | TwoThousandTo2100;
+
+/**
+ * Both 0 and 7 mean Sunday
+ */
+export type ZeroTo7 = 0 | 1 | 2 | 3 | 4 | 5 | 7;
+
+export type DayOfWeekName =
+  | 'SU'
```

Comment:
if we're going to have a separate type and stuff, why not make constants for these? then you just need one each

Reply:
I'm imagining string days being output from other libraries, or provided by end users, or `new Date().toString().substring(0,3)`, and wanting to cover all the cases.